dubbo-common/src/main/java/org/apache/dubbo/common/utils/UrlUtils.java

```diff
@@ -470,6 +470,16 @@ public static boolean isServiceKeyMatch(URL pattern, URL value) {
     }
 
     public static List<URL> classifyUrls(List<URL> urls, Predicate<URL> predicate) {
+        boolean filter = false;
+        for (URL url : urls) {
+            if (!predicate.test(url)) {
+                filter = true;
+                break;
+            }
+        }
+        if (!filter) {
+            return urls;

```

Comment:
keep an outer variable is dangerous.

Reply:
Better wrapping `urls` with unmodifiable?

Reply:
Let's keep this snippet as-is for this is not a critical path for performance improvement.